packages/core/src/codewhisperer/util/authUtil.ts

```diff
@@ -434,7 +434,7 @@ export class AuthUtil {
      * From the given connections, returns a connection that works with Amazon Q.
      */
     findUsableQConnection(connections: AwsConnection[]): AwsConnection | undefined {
-        const hasQScopes = (c: AwsConnection) => amazonQScopes.every(s => c.scopes?.includes(s))
+        const hasQScopes = (c: AwsConnection) => codeWhispererChatScopes.every(s => c.scopes?.includes(s))
```

Comment:
If this function returns a connection, do we know if it is working (non-expired). It is possible that we return a connection with the correct scopes but it is expired?

Reply:
Looks like `useConnection` (which this flow eventually calls) will validate the connection (see if it can retrieve a token) and then allow the user to proceed. So this doesn't look to be an issue